Device Agent - Raspbian Docu (#2906)
FlowFuse/flowfuse: docs/hardware/raspbian.md

---
navTitle: Raspberry Pi with Raspbian
navOrder: 3

> ---
> # Raspberry Pi
> 
> The Raspberry Pi section provides instructions for installing and setting up the FlowFuse Device Agent on your Raspberry Pi device, enabling seamless integration for efficient device management and automation.
> ## Installing the Device Agent

Reviewer: ```suggestion
---

# Raspberry Pi

The Raspberry Pi section provides instructions for installing and setting up the FlowFuse Device Agent on your Raspberry Pi device, enabling seamless integration for efficient device management and automation.

## Installing the Device Agent
```

> We provide a script to install Node.js, npm and the FlowFuse Device Agent onto a Raspberry Pi. This script won't work on ARMv6 builds as the standard Node.js builds don't support it, as result Pi Zero's are not supported.

Reviewer: ```suggestion
FlowFuse provides a script to install Node.JS, npm, and the FlowFuse Device Agent onto a Raspberry Pi.
This script won't work on ARMv6 builds as the standard Node.JS builds don't support it, as result Pi Zero's are not supported.
```

> 

Reviewer: ```suggestion
```

---

> navTitle: Raspberry Pi

Reviewer: ```suggestion
navTitle: Raspberry Pi with Raspbian
```

navOrder: 3
---
# Raspberry Pi

The Raspberry Pi section provides instructions for installing and setting up the FlowFuse Device Agent on your Raspberry Pi device, enabling seamless integration for efficient device management and automation.
## Installing the Device Agent

We provide a script to install Node.js, npm and the FlowFuse Device Agent onto a Raspberry Pi. This script won't work on ARMv6 builds as the standard Node.js builds don't support it, as result Pi Zero's are not supported.

> Running the following command will download and run the script. If you want to review the contents of the script first, you can view it [on Github](https://github.com/FlowFuse/device-agent/blob/main/service/raspbian-install-device-agent.sh).
> 
> #### One command installer

Reviewer: ```suggestion
```

```sh
bash <(curl -sL https://raw.githubusercontent.com/FlowFuse/device-agent/main/service/raspbian-install-device-agent.sh)
```

**This script will:**

1. Detect if Node.js is already installed, it will ensure it is at least v14. If less than v14 it will stop. If nothing is found it will install the Node.js 18 LTS release 
2. Install the latest version of the FlowFuse Device Agent using npm.
3. Setup the FlowFuse Device Agent to run as a service

## Running as a service

> Can can run the device agent as a service, this means it can run in the background and be enabled to automatically start on boot.

Reviewer: ```suggestion
You can run the device agent as a service, this means it can run in the background and be enabled to automatically start on boot.
```

But you just said this was done by the script? Why explain?

> ### Creating a Service File

Reviewer: Don't understand why this is moved though? It's not needed because of that script?

Author: I added more context, the commands can be useful for controlling the service or changing the default service settings.
